### PR TITLE
Add machine id to FUID

### PR DESCRIPTION
**Pull request by lchanmann**
## Ref
- https://nanopay.atlassian.net/browse/NP-5677

## Changes
- Add 2 bits machine id to FUID, default to the last two bytes of the host IP address MOD 255

### CODE REVIEW

File: src/foam/util/UIDGenerator.js [hunk 1]
```diff
@@ -26,6 +28,24 @@ foam.CLASS({
...
       name: 'salt',
       class: 'String'
+    },
+    {
+      name: 'machineId',
+      class: 'Int',
+      javaFactory: `
+        try {
+          return Integer.parseInt(System.getProperty("MACHINE_ID"));
+        } catch ( Exception e ) {
```
Comment (jlhughes) [inline src/foam/util/UIDGenerator.js:38]: This is going to be slow.

File: src/foam/util/UIDGenerator.js [hunk 2]
```diff
@@ -26,6 +28,24 @@ foam.CLASS({
...
+          try {
+            var ipAddress = InetAddress.getLocalHost();
+            var bytes     = ipAddress.getAddress();
+            var length    = bytes.length;
+            return (bytes[length - 1] & 0xff) +
+                   (bytes[length - 2] & 0xff) * (1<<8);
+          } catch ( UnknownHostException ex ) { /* Ignored */ }
```
Comment (jlhughes) [inline src/foam/util/UIDGenerator.js:45]: Think we should report - at least via System.err that we are unable to determine a machine id.

File: src/foam/util/UIDGenerator.js (hunk 1 above)
Comment (kgrgreer) [reply src/foam/util/UIDGenerator.js:38]: But we only call once, don't we?

Comment (jlhughes) [reply src/foam/util/UIDGenerator.js:38]: yes, you are correct.

**Pull request merged**